### PR TITLE
reference: update pessimistic transaction doc

### PR DESCRIPTION
**Pull request by coocood**
### What is changed, added or deleted? 
update pessimistic transaction doc for new version

### Which version does your change affect? 
dev, v3.0, v3.1

### CODE REVIEW

**Comment by coocood** [comment]
@jackysp PTAL

File: dev/reference/transactions/transaction-pessimistic.md [hunk 1]
```diff
@@ -48,29 +52,16 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
-
-相关配置都在 `[pessimistic-txn]` 类别下，除了前面介绍过的 `enable`，还可配置以下参数：
-
-- `ttl`
+## 和 MySQL innoDB 的差异
 
-    ```
-    ttl = "40s"
-    ```
+- TiDB 在使用范围做为 WHERE 条件，执行 DML 和 SELECT FOR UPDATE 语句时不会阻塞范围内并发的 INSERT 语句的执行。
```
[the remark this answers is not in the document]
Comment (juliezhang1112) [reply dev/reference/transactions/transaction-pessimistic.md]: Found one trailing space which leads to CI failure.

File: dev/reference/transactions/transaction-pessimistic.md [hunk 2]
```diff
@@ -48,29 +52,16 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
-    max-retry-count = 256
-    ```
-
-    悲观事务会在事务内自动重试单个语句，这个配置指定单个语句最大重试次数，设置这个参数是为了避免极端情况下，出现无限重试，无法停止。
-正常情况下不需要修改这个配置。
-
-## 已知的局限
-
-- 不支持 GAP Lock 和 Next Key Lock 在悲观事务内通过范围条件来更新多行数据的时候，其他的事务可以在这个范围内插入数据而不会被阻塞。
+TiDB 不支持 gap lock，也就不需要付出相应的并发性能的代价。
```
Comment (juliezhang1112) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
innoDB 通过实现 gap lock，支持阻塞 range 内并发的 INSERT 语句的执行，其主要目的是为了支持 statement based binlog，因此有些业务会通过降低隔离级别到 READ COMMITTED 来避免 gap lock 导致的并发性能问题。TiDB 不支持 gap lock，也就不需要付出相应的并发性能的代价。
```

Comment (juliezhang1112) [reply dev/reference/transactions/transaction-pessimistic.md]: Combine the three sentences into one paragraph and add some conjunctions to make the paragraph cohesive.

File: dev/reference/transactions/transaction-pessimistic.md [hunk 3]
```diff
@@ -48,29 +52,16 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
 
 - 不支持 SELECT LOCK IN SHARE MODE。
```
Comment (juliezhang1112) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- TiDB 不支持 SELECT LOCK IN SHARE MODE。
```

Comment (juliezhang1112) [reply dev/reference/transactions/transaction-pessimistic.md]: Pointing out the subject "TiDB" makes the sentence clearer.

**Comment by juliezhang1112** [comment]
LGTM

File: dev/reference/transactions/transaction-pessimistic.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
 
 ## 悲观事务模式的行为
 
-悲观事务的行为和 MySQL 基本一致（不一致的地方详见[已知的局限](#已知的局限)）：
+悲观事务的行为和 MySQL 基本一致（不一致的地方详见[和 MySQL innoDB 的差异](#和-mysql-innodb-的差异)）：
```
Comment (lilin90) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 5]
```diff
@@ -22,10 +22,14 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
 
 - 事务提交或回滚的时候，会释放所有的锁。
 
-- 如果并发事务出现死锁，会被死锁检测器检测到，并返回和 MySQL 相同的 DEADLOCK 错误。
+- 如果并发事务出现死锁，会被死锁检测器检测到，并返回兼容 MySQL 的错误码 1213。
 
 - 乐观事务和悲观事务可以共存，事务可以任意指定使用乐观模式或悲观模式来执行。
 
+- 通过设置 @@innodb_wait_timeout 变量，设置等锁超时时间，等锁超时后返回兼容 MySQL 的错误码 1205。
```
Comment (lilin90) [inline dev/reference/transactions/transaction-pessimistic.md]: Can we show it this way?

```suggestion
- 通过设置 `innodb_wait_timeout` 变量，设置等锁超时时间，等锁超时后返回兼容 MySQL 的错误码 1205。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 6]
```diff
@@ -48,29 +52,12 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
-    ```
-
-    `ttl` 是悲观事务的锁超时时间，默认值 40 秒，可以在 15 秒到 120 秒之间配置，超出这个范围会报错。如果事务的执行时间超过了 `ttl`，事务会失败。设置过大，会存在 `tidb-server` 宕机时，残留的悲观锁长时间阻塞写的问题，设置的过小，会存在事务来不及执行完，就被其他事务 rollback 的问题。
-
-- `max-retry-count`
-
-    ```
-    max-retry-count = 256
-    ```
+## 和 MySQL innoDB 的差异
```
Comment (lilin90) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
## 和 MySQL InnoDB 的差异
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 7]
```diff
@@ -22,10 +22,14 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
 - 事务提交或回滚的时候，会释放所有的锁。
 
-- 如果并发事务出现死锁，会被死锁检测器检测到，并返回和 MySQL 相同的 DEADLOCK 错误。
+- 如果并发事务出现死锁，会被死锁检测器检测到，并返回兼容 MySQL 的错误码 1213。
 
 - 乐观事务和悲观事务可以共存，事务可以任意指定使用乐观模式或悲观模式来执行。
 
+- 通过设置 @@innodb_wait_timeout 变量，设置等锁超时时间，等锁超时后返回兼容 MySQL 的错误码 1205。
+
+- 支持 FOR UPDATE NOWAIT 语法，遇到锁时不会阻塞等锁，而是返回兼容 MySQL 的错误码 3572。
```
Comment (lilin90) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- 支持 `FOR UPDATE NOWAIT` 语法，遇到锁时不会阻塞等锁，而是返回兼容 MySQL 的错误码 3572。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 8]
```diff
@@ -48,29 +52,12 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
-- `max-retry-count`
-
-    ```
-    max-retry-count = 256
-    ```
+## 和 MySQL innoDB 的差异
 
-    悲观事务会在事务内自动重试单个语句，这个配置指定单个语句最大重试次数，设置这个参数是为了避免极端情况下，出现无限重试，无法停止。
-正常情况下不需要修改这个配置。
+1. TiDB 使用 range 作为 WHERE 条件，执行 DML 和 SELECT FOR UPDATE 语句时不会阻塞范围内并发的 INSERT 语句的执行。
```
Comment (lilin90) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
1. TiDB 使用 range 作为 WHERE 条件，执行 DML 和 `SELECT FOR UPDATE` 语句时不会阻塞范围内并发的 `INSERT` 语句的执行。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 9]
```diff
@@ -48,29 +52,12 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
 
-## 已知的局限
+    innoDB 通过实现 gap lock，支持阻塞 range 内并发的 INSERT 语句的执行，其主要目的是为了支持 statement based binlog，因此有些业务会通过降低隔离级别到 READ COMMITTED 来避免 gap lock 导致的并发性能问题。TiDB 不支持 gap lock，也就不需要付出相应的并发性能的代价。
```
Comment (lilin90) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
    InnoDB 通过实现 gap lock，支持阻塞 range 内并发的 `INSERT` 语句的执行，其主要目的是为了支持 statement based binlog，因此有些业务会通过将隔离级别降低至 READ COMMITTED 来避免 gap lock 导致的并发性能问题。TiDB 不支持 gap lock，也就不需要付出相应的并发性能的代价。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 10]
```diff
@@ -48,29 +52,12 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
 
-- 不支持 GAP Lock 和 Next Key Lock 在悲观事务内通过范围条件来更新多行数据的时候，其他的事务可以在这个范围内插入数据而不会被阻塞。
+2. TiDB 不支持 SELECT LOCK IN SHARE MODE。
```
Comment (lilin90) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
```

**Comment by lilin90** [comment]
@coocood What if users are using 3.0.1 ~ 3.0.5? And how to understand 3.1.6 since we haven't released 3.1.0? Perhaps we need to make it clearer. @juliezhang1112 FYI.

File: v3.0/reference/transactions/transaction-pessimistic.md [hunk 11]
```diff
@@ -10,22 +10,24 @@ TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败
...
 - 当一行数据被加了悲观锁以后，其他尝试读取这一行的事务不会被阻塞，可以读到已提交的数据。
 
 - 事务提交或回滚的时候，会释放所有的锁。
 
-- 如果并发事务出现死锁，会被死锁检测器检测到，并返回和 MySQL 相同的 DEADLOCK 错误。
+- 如果并发事务出现死锁，会被死锁检测器检测到，并返回兼容 MySQL 的错误码 1213。
 
 - 乐观事务和悲观事务可以共存，事务可以任意指定使用乐观模式或悲观模式来执行。
 
+- 从 TiDB 3.0.6 开始，支持通过设置 `innodb_wait_timeout` 变量，设置等锁超时时间，等锁超时后返回兼容 MySQL 的错误码 1205。
```
Comment (lilin90) [inline v3.0/reference/transactions/transaction-pessimistic.md]: To make it clearer with our version mark.

```suggestion
- 支持通过设置 `innodb_wait_timeout` 变量 <span class="version-mark">从 v3.0.6 开始引入</span>，设置等锁超时时间，等锁超时后返回兼容 MySQL 的错误码 `1205`。
```

**Comment by juliezhang1112** [comment]
> @coocood What if users are using 3.0.1 ~ 3.0.5? And how to understand 3.1.6 since we haven't released 3.1.0? Perhaps we need to make it clearer. @juliezhang1112 FYI.

1. Remove the typo `3.1.6` in the PR description.
2. The `innodb_wait_timeout` variable was introduced in TiDB 3.0.6 or later versions. Therefore, for users using 3.0.1 ~ 3.0.5, the `innodb_wait_timeout` variable is not supported. Apart from the variable, the document applies to all the versions of TiDB 3.0.

**Comment by lilin90** [comment]
@juliezhang1112 Got it. Thanks! And these changes apply to `dev` and all versions of `v3.1`.

**Pull request merged**